TMDM-11494 Page info is incorrect when entity's FK has multiple values.

yjli-talend opened the pull request:
https://jira.talendforge.org/browse/TMDM-11494

**What is the current behavior?**
When entity has multiple foreign keys, we get wrong count number of rows. For instance, we have one record pointing to three referenced records, we display three rows in 'browserecords' list (denormalization), but we have a total number of just one row, which causes issue for paging widget.

**What is the new behavior?**
When entity has multiple foreign keys, in order to get the correct record count, we need to add join to count select. We can get correct total rows number.
                  


**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...


**Other information**:

pbabin commented on org.talend.mdm.core.storage.sql/src/test/java/com/amalto/core/query/DataRecordCreationTest.java:
```diff
@@ -93,6 +107,54 @@ public void __testCompositeKeyAndFK() {
+        MetadataRepository repository = new MetadataRepository();
+        repository.load(DataRecordCreationTest.class.getResourceAsStream("MultipleForeignKey.xsd")); //$NON-NLS-1$
+
+        Storage storage = new HibernateStorage("H2-DS1", StorageType.MASTER); //$NON-NLS-1$
```
Since there is NLS suppression annotation on the test class, $NON-NLS-$ is useless

pbabin commented on org.talend.mdm.core.storage.sql/src/test/java/com/amalto/core/query/DataRecordCreationTest.java:
```diff
@@ -93,6 +107,54 @@ public void __testCompositeKeyAndFK() {
+        DataRecordReader<String> factory = new XmlStringDataRecordReader();
+
+        List<DataRecord> records = new LinkedList<DataRecord>();
+        records.add(factory.read(repository, repository.getComplexType("FK1"), "<FK1><Id>FK1</Id><Name>FKInfo1</Name></FK1>")); //$NON-NLS-1$//$NON-NLS-2$
```
Bad naming choice leading to confusion when reading, would have been better to use something like below.
It seems also that the 3rd test is useless, exactly same as the second test.
```
<xsd:element` name="Child">
  <xsd:complexType>
    <xsd:all>
      <xsd:element maxOccurs="1" minOccurs="1" name="Id" type="xsd:string" />
      <xsd:element maxOccurs="1" minOccurs="0" name="Name" type="xsd:string" />
    </xsd:all>
  </xsd:complexType>
</xsd:element>
```
```
<Child><Id>1</Id><Name>Child1</Name></Child>
<Child><Id>2</Id><Name>Child2</Name></Child>
<Child><Id>3</Id><Name>Child3</Name></Child>
```
```
<xsd:element name="Parent1">
  <xsd:complexType>
    <xsd:all>
      <xsd:element maxOccurs="1" minOccurs="1" name="Id" type="xsd:string" />
      <xsd:element maxOccurs="1" minOccurs="0" name="Name" type="xsd:string" />
      <xsd:element maxOccurs="unbounded" minOccurs="0" name="ChildFK" type="xsd:string" />
    </xsd:all>
  </xsd:complexType>
</xsd:element>
```
```
<Parent1>
  <Id>1</Id>
  <Name>Parent1</Name>
  <ChildFK>[1]</ChildFK>
  <ChildFK>[2]</ChildFK>
  <ChildFK>[3]</ChildFK>
</Parent1>
```
```
<xsd:element name="Parent2">
  <xsd:complexType>
    <xsd:all>
      <xsd:element maxOccurs="1" minOccurs="1" name="Id" type="xsd:string" />
      <xsd:element maxOccurs="1" minOccurs="0" name="Name" type="xsd:string" />
      <xsd:element maxOccurs="unbounded" minOccurs="0" name="SonFK" type="xsd:string" />
      <xsd:element maxOccurs="unbounded" minOccurs="0" name="DaughterFK" type="xsd:string" />
      <xsd:element maxOccurs="unbounded" minOccurs="0" name="Names" type="xsd:string">
    </xsd:all>
  </xsd:complexType>
</xsd:element>
```
```
<Parent2>
  <Id>2</Id>
  <Name>Parent2</Name>
  <SonFK>[1]</SonFK>
  <SonFK>[2]</SonFK>
  <SonFK>[3]</SonFK>
  <DaughterFK>[1]</DaughterFK>
  <DaughterFK>[2]</DaughterFK>
  <DaughterFK>[3]</DaughterFK>
  <Names>Names1</Names>
  <Names>Names2</Names>
</Parent1>
```

yjli-talend commented:
About test entity 3,it is different from entity 2.The foreign key of entity 2 point to foreign entity.And The foreign key of entity 3 point to key of foreign entity.I have modified entity 2,so entity 3 is unnecessary now and I removed it.
